package io.github.tufein.duofrost

import io.github.tufein.duofrost.animations.FadeTransitionAnimation
import io.github.tufein.duofrost.animations.LedAnimationType
import io.github.tufein.duofrost.tools.PerformanceProfile

data class LedPreset(
    val name: String,
    val animationType: LedAnimationType,
    val performanceProfile: PerformanceProfile,
    val color: Int,
    val rightColor: Int = color,
    val fadeEndColor: Int = FadeTransitionAnimation.DEFAULT_END_COLOR,
    val fadeEndRightColor: Int = fadeEndColor,
    val brightness: Int,
    val speed: Float,
    val smoothness: Float,
    val sensitivity: Float = 0.5f,
    val saturationBoost: Float = 0.0f,
    val useCustomSampling: Boolean = false,
    val useSingleColor: Boolean = false,
    val breatheWhenCharging: Boolean = false,
    val indicateChargingSpeed: Boolean = false,
    val flashWhenReady: Boolean = false,
    val batteryLowColorOverride: Int? = null,
    val batteryMidColorOverride: Int? = null,
    val batteryHighColorOverride: Int? = null,
    val cpuCoolColorOverride: Int? = null,
    val cpuWarmColorOverride: Int? = null,
    val cpuHotColorOverride: Int? = null,
    val isAppProfileDefault: Boolean = false,
    val ragnarokAccepted: Boolean = false,
    val icon: PresetIcon = PresetIcon.LIGHT,
    val customEmoji: String? = null,
    val customImageFileName: String? = null,
    val appIconPackageName: String? = null,
    val ownerPackage: String? = null
)